src/main/scala/com/kinja/amqp/AmqpConsumer.scala

```diff
@@ -72,6 +107,14 @@ class AmqpConsumer(
 							sender ! Reject(envelope.getDeliveryTag, requeue = false)
 					}
 			}
+
+			def asleep(originalSender: ActorRef, ack: Ack): Receive = {
+				case WakeUp =>
+					originalSender ! ack
+					context.unbecome()
+				case Delivery(_, envelope, _, _) =>
+					sender ! Reject(envelope.getDeliveryTag, requeue = true)
+			}
```

Comment:
I tried subscribing to the shutdown event here so that we immediately send the ack when when shutdown is initiated but I couldn't find a way to deliver that message to this actor without rewriting the client library. That's why now this case is simply treated as if processing didn't finish before shutting down the app, the ack will be lost and processing will restart on another node.